### PR TITLE
feat(api,cli): import, export action

### LINKED ISSUE #2273: feat (cdsctl): cdsctl action add
to implement (available in old cli for now)

### PR DESCRIPTION
**Pull request by fsamin**
close #2273

### CODE REVIEW

File: engine/api/action/action_exporter.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	"github.com/go-gorp/gorp"
+	"github.com/ovh/cds/sdk"
+	"github.com/ovh/cds/sdk/exportentities"
+)
+
+// Export an action
+func Export(db gorp.SqlExecutor, name string, f exportentities.Format, u *sdk.User, w io.Writer) (int, error) {
+	a, err := LoadPublicAction(db, name)
+	if err != nil {
+		return 0, sdk.WrapError(sdk.ErrNotFound, "Export> Cannot load action")
```
Comment (bnjjj) [inline engine/api/action/action_exporter.go]: display original error in log

File: engine/api/action/action_exporter.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package action
+
+import (
+	"io"
+
+	"github.com/go-gorp/gorp"
+	"github.com/ovh/cds/sdk"
```
Comment (sguiheux) [inline engine/api/action/action_exporter.go]: empty line please

File: engine/api/action/action_exporter.go (hunk 1 above)
Comment (sguiheux) [inline engine/api/action/action_exporter.go]: err, LoadPublicAction must return errNotFound if no rows returned

File: cli/cdsctl/action.go [hunk 3]
```diff
@@ -89,3 +94,54 @@ func actionDocRun(v cli.Values) error {
 	fmt.Println(sdk.ActionInfoMarkdown(action, path.Base(v["path"])))
 	return nil
 }
+
+var actionImportCmd = cli.Command{
+	Name:  "import",
+	Short: "Import a CDS action",
```
Comment (yesnault) [inline cli/cdsctl/action.go]: Example: `Import an action
cds action import --path $HOME/src/github.com/ovh/cds/contrib/actions/action-scripts/cds-docker-package.hcl
cds action import --path https://raw.githubusercontent.com/ovh/cds/master/contrib/actions/cds-docker-package.hcl 
`

Import a url if ok too ? (as old cli)

File: sdk/exportentities/action.go [hunk 4]
```diff
@@ -404,3 +436,36 @@ func (s Step) IsFlagged(flag string) (bool, error) {
...
+		param := sdk.Parameter{
+			Name:        p,
+			Type:        v.Type,
+			Value:       v.DefaultValue,
+			Description: v.Description,
+		}
+		if param.Type == "" {
+			param.Type = sdk.StringParameter
+		}
+		a.Parameters = append(a.Parameters, param)
```
Comment (bnjjj) [inline sdk/exportentities/action.go]: we can initialise with a make(..., len...)

File: cli/cdsctl/action.go [hunk 5]
```diff
@@ -89,3 +94,54 @@ func actionDocRun(v cli.Values) error {
...
+
+func actionImportRun(v cli.Values) error {
+	path := v.GetString("path")
+	f, err := os.Open(path)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	var format = "yaml"
```
Comment (yesnault) [inline cli/cdsctl/action.go]: no HCL??

File: cli/cdsctl/action.go [hunk 6]
```diff
@@ -89,3 +94,54 @@ func actionDocRun(v cli.Values) error {
...
+	}
+
+	fmt.Printf("%s successfully imported\n", path)
+
+	return client.ActionImport(f, format)
+}
+
+var actionExportCmd = cli.Command{
+	Name:  "export",
+	Short: "Export a CDS action",
```
Comment (yesnault) [inline cli/cdsctl/action.go]: Example

File: sdk/cdsclient/client_action.go [hunk 7]
```diff
@@ -32,3 +37,44 @@ func (c *client) ActionList() ([]sdk.Action, error) {
...
+	url := "/action/import"
+	mods := []RequestModifier{}
+	switch format {
+	case "json":
+		mods = []RequestModifier{
+			func(r *http.Request) {
+				r.Header.Set("Content-Type", "application/json")
+			},
+		}
+	case "yaml", "yml":
```
Comment (yesnault) [inline sdk/cdsclient/client_action.go]: HCL?

File: cli/cdsctl/action.go (hunk 3 above)
Comment (fsamin) [reply cli/cdsctl/action.go]: nop

File: cli/cdsctl/action.go (hunk 5 above)
Comment (fsamin) [reply cli/cdsctl/action.go]: nop

File: sdk/cdsclient/client_action.go (hunk 7 above)
Comment (fsamin) [reply sdk/cdsclient/client_action.go]: nop

File: cli/cdsctl/action.go (hunk 6 above)
Comment (fsamin) [reply cli/cdsctl/action.go]: done

**Pull request merged**